config.go

```diff
@@ -551,31 +551,60 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
-func (s *Settings) GetGithubOauthString() (string, error) {
+func (s *Settings) GetGithubOauthString() ([]string, error) {
+	var tokens []string
+	var token_name string
+
 	token, ok := s.Credentials["github"]
 	if ok && token != "" {
-		return token, nil
+		// we want to make sure tokens[0] is always the default token
+		tokens = append(tokens, token)
+	} else {
+		return nil, errors.New("no 'github' token in settings")
 	}
 
-	return "", errors.New("no github token in settings")
+	for i := 1; i < 10; i++ {
+		token_name = fmt.Sprintf("github_alt%d", i)
+		token, ok := s.Credentials[token_name]
+		if ok && token != "" {
+			tokens = append(tokens, token)
+		} else {
+			break
+		}
+	}
+	return tokens, nil
 }
 
 func (s *Settings) GetGithubOauthToken() (string, error) {
 	if s == nil {
 		return "", errors.New("not defined")
 	}
 
-	oauthString, err := s.GetGithubOauthString()
+	oauthStrings, err := s.GetGithubOauthString()
 	if err != nil {
 		return "", err
 	}
+	timeSeed := time.Now().Nanosecond()
+	for i, oauthString := range oauthStrings {
```

Comment:
I get why we're randomizing kind of but the process of it is a bit confusing to me; is it possible we could just get stuck in like an infinite loop? And if there's an error with one of the oauth strings, should we just try another one?

Reply:
The modulo arithmetic works out from first principles, and I ran through line-by-line with a couple examples. We are guaranteed to choose one and only one token per instance of `oauthStrings`. 

There are n distinct modulo equivalence classes for modulus n. For example, there are two modulo equivalence classes for mod 2: {0,2,4...} and {1,3,5...}. In any loop, we advance through n consecutive integers, where n is the length of oauthStrings. Each time we iterate, we enter a new equivalence class. So, by something like the pigeonhole principle, we have to land on the equivalence class of {0,n,2n,3n...} once and only once. 
Say we get a `timeSeed` that is in the equivalence class with n+1. Then, on the final iteration, `randomOffset = (n+1) + (n-1)`, or 2n. 2n is in the equivalence class with 0.

Reply:
I like the idea of just trying another one if there's an error. I'll code that up

Reply:
Thank you for that walkthrough!